File: 70-climbing-stairs/climbing-stairs.cpp
class Solution {
public:
    int climb (int n,int dp[])
    {
        if(n==1) return 1;
        if(n==2) return 2;
        if(dp[n]!= -1) return dp[n];
        
        dp[n]=climb(n-1,dp)+climb(n-2,dp);
        return dp[n];
    }

    int climbStairs(int n) {
        int dp[100];
        for(int i=0;i<=n;i++)
        {
            dp[i]=-1;
        }
        return climb(n,dp);
    }
};

File: 70-climbing-stairs/climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        """
        :type n: int
        :rtype: int
        """
        res=0
        memo={}
        def solve(steps):
            
            if steps ==  0:
                return 1
            if steps < 0:
                return 0
            if steps in memo:
                return memo[steps]
            memo[steps]=solve(steps-1) + solve(steps-2)
            return memo[steps]
        return solve(n)